lms/djangoapps/courseware/tests/test_model_data.py

```diff
@@ -139,7 +139,7 @@ def test_set_existing_field(self):
         # to discover if something other than the DjangoXBlockUserStateClient
         # has written to the StudentModule (such as UserStateCache setting the score
         # on the StudentModule).
-        with self.assertNumQueries(2, using='default'):
+        with self.assertNumQueries(4, using='default'):
```

Comment:
@ormsbee Please take a look. This fix would add a begin/end savepoint around every `student_module.save()`, as shown by the test modifications.